src/lib.rs

```diff
@@ -954,6 +965,43 @@ impl Build {
         self
     }
 
+    /// Backward compatible version of [`Build::cpp_link_stdlib_path`]
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// cc::Build::new()
+    ///     .file("src/foo.c")
+    ///     .shared_flag(true)
+    ///     .cpp_link_stdlib("stdc++")
+    ///     .compile("libfoo.so");
+    /// ```
+    #[deprecated(since = "1.0.100", note = "please use `cpp_link_stdlib_path` instead")]
+    pub fn cpp_link_stdlib<'a, V: Into<Option<&'a str>>>(
```

Comment:
Do we absolutely have to deprecate this rather than, I dunno, doing some special trait shenanigans so it works with str but OsStr also works? Maybe we do but it seems unfortunate.

Reply:
No I don't think we can, `Into< Option<&str>>` prevents this.

Reply:
Yeah, this is unfortunate. It's usually not a path, but a literal `"c++"` or `"stdc++"` so I don't know that deprecating this is worth it.

Reply:
I will revert that

Reply:
Reverted

Reply:
I've reverted changes to `cpp_{set, link}_stdlib` according to review feedback of @thomcc